src/providers/ogr/qgsogrexpressioncompiler.cpp

```diff
@@ -43,6 +43,8 @@ QgsSqlExpressionCompiler::Result QgsOgrExpressionCompiler::compile( const QgsExp
     return Fail;
   else if ( mSource->mDriverName == "MSSQLSpatial" )
     return Fail;
+  else if ( mSource->mDriverName == "GPKG" )
```

Comment:
This seems like a real shame... the other drivers failing isn't a big issue because they generally already have a dedicated provider. But having no compilation support for this important format is a loss. 

Could we re-use the spatialite expression compiler in this case? The syntax should be the same.